code/game/objects/items/devices/multitool.dm

```diff
@@ -48,11 +56,19 @@
 	if(buffer)
 		. += "<span class='notice'>Its buffer contains [buffer].</span>"
 
-/obj/item/multitool/attack_self(mob/user)
+/obj/item/multitool/polytool/attack_self(mob/user)
+	playsound(get_turf(user), 'sound/items/change_jaws.ogg', 50, 1)
+	var/obj/item/t_scanner/polytool/polytray = new /obj/item/t_scanner/polytool(drop_location())
+	to_chat(user, "<span class='notice'>You change polytool mode to T-Ray.</span>")
+	qdel(src)
```

Comment:
To wygląda jakby opróżniało bufor multitoola przy każdej zmianie trybu. Wiem, że jest to problem związany z sposobem w który zostały zakodowane narzędzia z różnymi trybami pracy, więc może w przyszłości przydałby się kompletny refactor ich kodu, ale na razie może warto dodać zmienną przechowującą bufor multitoola gdy jest w trybie T-ray?

Reply:
To prawda, opróżnia, ale nie miałem pomysłu jak to rozwiązać prawdę mówiąc, a wydawało mi się, że to nie jest duży problem. "Fajnie by było", ale gładki mózg robi brrr.

Reply:
Najszybszym rozwiązaniem byłoby dodanie zmiennej, np "var/obj/machinery/multibuffer" do klasy poly t-raya i polycutterów, a attack_self() naszego multitoola zmienić na coś w stylu 
```
/obj/item/multitool/polytool/attack_self(mob/user)
	playsound(get_turf(user), 'sound/items/change_jaws.ogg', 50, 1)
	var/obj/item/t_scanner/polytool/polytray = new /obj/item/t_scanner/polytool(drop_location())
	to_chat(user, "<span class='notice'>You change polytool mode to T-Ray.</span>")
	polytray.multibuffer = buffer //nasza zmiana
	qdel(src)
	user.put_in_active_hand(polytray)
```
Oraz w attack_self() T-raya przed qdelem dodać coś w stylu
```
	polycutter.multibuffer = multibuffer
```
I na samym końcu w attack_self() polycutterów przed qdelem dodać
```
	polymulti.buffer = multibuffer
```